File: Assets/Scripts/Ant.js
var jumpSpeed: float = 9; // vertical jump initial speed
var forwardJumpFactor: float = 0.2;

private var anim: Animator;
private var rb: Rigidbody;
private var wallWalker: WallWalker;

function Start() {
	rb = GetComponent.<Rigidbody>();
  anim = GetComponent.<Animator>();
  wallWalker = GetComponent('WallWalker');
}

function FixedUpdate() {
  // Perform jump
  if (Input.GetButtonDown('Jump')) {
    if (wallWalker.isGrounded) {
      // Forward motion takes away from max jump height
      rb.velocity += (jumpSpeed) * transform.up;

      // Add forward momentum
      // Bug: Backwards momentum seems to be too much
      // Bug: Slingshots off of hills
      // Bug: Jumps too high randomly
      rb.velocity += wallWalker.forwardMotion * forwardJumpFactor * transform.forward;
    }
  }

	// Set animation parameters
	anim.SetBool('attack', Input.GetButtonDown('Fire1'));
	anim.SetBool('attack2', Input.GetButtonDown('Fire2'));
	anim.SetBool('attack3', Input.GetButtonDown('Fire3'));
}

File: Assets/Scripts/Ant.cs
using System;
using UnityEngine;
using System.Collections;
using UnityStandardAssets.CrossPlatformInput;

public class Ant : WallWalker {
  /** Configuration */
  public float jumpSpeed = 9f; // vertical jump initial speed
  public float forwardJumpFactor = 0.2f;

  /** API */
  public bool isJumping = false;

  protected override void FixedUpdate() {
    // Run WallWalker calculations so we get isGroudned and inputs
    base.calculate();

    // Be sticky no matter what
    base.beSticky();

    base.wallWalk();

    // Perform jump
    if (isGrounded) {
      if (!isJumping && ((throttleStick != "" && CrossPlatformInputManager.GetAxis(throttleStick) > 0) || CrossPlatformInputManager.GetButton("Jump"))) {
        isJumping = true;
        StartCoroutine(jump());
      }
    }

  	// Set animation parameters
  	anim.SetBool("attack", CrossPlatformInputManager.GetButtonDown("Fire1"));
  	anim.SetBool("attack2", CrossPlatformInputManager.GetButtonDown("Fire2"));
  	// anim.SetBool("attack3", CrossPlatformInputManager.GetButtonDown("Fire3"));
  }

  protected IEnumerator jump() {
    // Forward motion takes away from max jump height
    rb.velocity += (jumpSpeed) * transform.up;

    // Add forward momentum
    rb.velocity += forwardMotion * forwardJumpFactor * transform.forward;

    gravity -= jumpSpeed/2;

    // Don't allow jumping for a bit
    yield return new WaitForSeconds(0.5f);

    gravity += jumpSpeed/2;

    isJumping = false;
  }
}
